syslog-ng-rs/syslog-ng-common/src/proxies/parser/mod.rs

```diff
@@ -117,20 +117,31 @@ pub mod _parser_plugin {
 
     #[no_mangle]
     pub extern fn native_parser_proxy_process(this: &mut ParserProxy<$name>, parent: *mut $crate::sys::LogParser, msg: *mut $crate::sys::LogMessage, input: *const c_char) -> c_int {
-        let input = unsafe { CStr::from_ptr(input).to_str() };
-        let mut parent = LogParser::wrap_raw(parent);
-        let mut msg = LogMessage::wrap_raw(msg);
-        let result = match input {
-            Ok(input) => this.process(&mut parent, &mut msg, input),
-            Err(err) => {
-                error!("{}", err);
-                false
+        let mut wrapper_this = AssertUnwindSafe(this);
+        let wrapper_parent = AssertUnwindSafe(parent);
```

Comment:
I suppose `*mut` is a raw pointer, it doesn't provide any guarantee. `LogParser::wrap_raw()` wraps this into a higher level type, which needs to be `mut`.